File: November Dawn 2020/Yash's Password/yash-password.cpp
#include <bits/stdc++.h>

using namespace std;
int main()
{
    string s;
    int l;
    cin>>l;
    cin>>s;
    int point1=0,point2=0,point3=0,point4=0,flag,count=0,need;
    string numbers = "0123456789";
    string lower_case = "abcdefghijklmnopqrstuvwxyz";
    string upper_case = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    string special_characters = "!@#$%^&*()-+";
    for(int i=0;i<s.length();i++)
    {   
        flag=0;
        for(int j=0;j<numbers.length();j++)
            if(s[i]==numbers[j])
            {
                point1++;
                flag=1;
                break;
            }
            if(flag == 1)
                continue;
        for(int j=0;j<lower_case.length();j++)
            if(s[i]==lower_case[j])
            {
                point2++;
                flag=1;
                break;
            }
            if(flag == 1)
                continue;
            for(int j=0;j<upper_case.length();j++)
            if(s[i]==upper_case[j])
            {
                point3++;
                flag=1;
                break;
            }
            if(flag == 1)
                continue;
            for(int j=0;j<special_characters.length();j++)
            if(s[i]==special_characters[j])
            {
                point4++;
                flag=1;
                break;
            }
            if(flag == 1)
                continue;    
    }
    if(point1==0)
        count++;
    if(point2==0)
        count++;
    if(point3==0)
        count++;
    if(point4==0)
        count++;
    if(l>6)
    cout<<count;
    else
    {
        need =7-l;
        int max = need>count ? need : count;
        cout<<max;
    }



}

File: November Dawn 2020/Yash's Password/yash-password.java
import java.util.*;
public class Solution
{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        int l = sc.nextInt();
        String s = sc.next();

        int p1 = 0, p2 = 0, p3 = 0, p4 = 0, count = 0;
        int flag = 0, need = 0;

        String numbers = "0123456789";
        String lower_case = "abcdefghijklmnopqrstuvwxyz";
        String upper_case = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        String special_characters = "!@#$%^&*()-+";

        for(int i=0;i<s.length();i++)
        {
            flag = 0;
            for(int j=0;j < numbers.length(); j++)
            {
                if(s.charAt(i) == numbers.charAt(j))
                {
                    p1++;
                    flag = 1;
                    break;
                }
            }
            if(flag == 1)
                continue;

            for(int j=0;j < lower_case.length();j++)
            {
                if(s.charAt(i) == lower_case.charAt(j))
                {
                    p2++;
                    flag = 1;
                    break;
                }
            }
            if(flag == 1)
                continue;

            for(int j=0;j < upper_case.length();j++)
            {
                if(s.charAt(i) == upper_case.charAt(j))
                {
                    p3++;
                    flag = 1;
                    break;
                }
            }
            if(flag == 1)
                continue;

            for(int j=0;j < special_characters.length();j++)
            {
                if(s.charAt(i) == special_characters.charAt(j))
                {
                    p4++;
                    flag = 1;
                    break;
                }
            }
            if(flag == 1)
                continue;
        }

        if(p1 == 0)
            count++;
        if(p2 == 0)
            count++;
        if(p3 == 0)
            count++;
        if(p4 == 0)
            count++;

        if(l > 6)
            System.out.println(count);
        else
        {
            need = 7-l;
            int max = (need > count)? need: count;
            System.out.println(max);
        }

    }
}